homeassistant/components/mhtzn/Gateway.py

```diff
@@ -0,0 +1,200 @@
+        elif topic.endswith("p28"):
+            """Scene List data"""
+            scene_list = payload["data"]
+            for scene in scene_list:
+                scene["unique_id"] = f"{scene['id']}"
+                await self._add_entity("scene", scene)
+        elif topic.endswith("event/3"):
+            """Device state data"""
+            stats_list = payload["data"]
+            for state in stats_list:
+                async_dispatcher_send(
+                    self._hass, "mhtzn_device_state_{}".format(state["sn"]), state
+                )
+        elif topic.endswith("p33"):
+            """Basic data, including room information, light group information, curtain group information"""
+            for room in payload["data"]["rooms"]:
+                self.room_map[room["id"]] = room
+            for lightGroup in payload["data"]["lightsSubgroups"]:
+                self.light_group_map[lightGroup["id"]] = lightGroup
+        elif topic.endswith("p31"):
+            """Relationship data for rooms and groups"""
+            for room in payload["data"]:
+                room_id = room["room"]
+                room_name = "默认房间"
+                if room_id == 0:
+                    room_name = "全屋"
+                elif room_id in self.room_map:
+                    room_instance = self.room_map[room_id]
+                    room_name = room_instance["name"]
+
+                for light_group_id in room["lights"]:
+                    device_name = "默认灯组"
+                    if light_group_id == 0:
+                        device_name = "所有灯"
+                    elif light_group_id in self.light_group_map:
+                        light_group = self.light_group_map[light_group_id]
+                        device_name = light_group["name"]
+
+                    group = {
+                        "unique_id": f"{room_id}-{light_group_id}",
+                        "room": room_id,
+                        "subgroup": light_group_id,
+                        "is_group": True,
+                        "name": f"{room_name}-{device_name}",
+                    }
+                    await self._add_entity("light", group)
+
+    async def _add_entity(self, component: str, device: dict):
+        """Add child device information"""
+        async_dispatcher_send(
+            self._hass, EVENT_ENTITY_REGISTER.format(component), device
+        )
+
+    async def reconnect(self, entry: ConfigEntry):
+        """Reconnect gateway MQTT"""
+        mqtt_client: MQTT = self._hass.data[MQTT_CLIENT_INSTANCE]
+        mqtt_client.conf = entry.data
+        await mqtt_client.async_disconnect()
+        mqtt_client.init_client()
+        await mqtt_client.async_connect()
+
+    async def init(self):
+        """Initialize the gateway business logic, including subscribing to device data, scene data, and basic data,
+        and sending data reporting instructions to the gateway"""
+        discovery_topics = [
+            # 获取设备
```

Comment:
Aläl comment should be in English.

Reply:
Thank you for your hint, I have changed it to English, please review it again

Reply:
> Aläl comment should be in English.

Thank you for your hint, I have changed it to English, please review it again

Reply:
All update request processed